app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                $postMetaSkw->keyword_id = $subKeyword->id;
+                $postMetaSkw->save();
+            }
+
+
+        }
+
+        if(strpos($request->get('submit_type'),'公開') !== false || strpos($request->get('submit_type'),'下書き') !== false){
+            return redirect()->to(action('Webapp\\PostController@getEdit', $post->id))->send();
+        }else if(strpos($request->get('submit_type'),'プレビュー') !== false){
+            return redirect()->to(action('Webapp\\PostController@getPreviewIframe', $post->id))->send();
+        }
+
+        return redirect()->to(action('Webapp\\PostController@getEdit', $post->id))->send();
+    }
+
+    public function postCreateAjax(Request $request) {
+        $post = new Post();
+
+        if(($postId = $request->get('post_id', 0)) != 0) {
+            $post = (!empty($postTmp = Post::find($postId)) && $postTmp->user_id == $this->currentUser->id) ? $postTmp : $post;
+        }
+
+        $postTitle = $request->get('post_title');
+        $postContent = $request->get('post_content');
+        $postChecksum = $request->get('checksum', 0);
+        $postDescription = trim($request->get('post_description'));
+
+        if($post->isLock($this->currentSession)) {
+            $this->activityLog->setInfo('Locked');
+            return ['error' => [
+                'message' => 'このコンテンツは現在ロックされています。編集できません'
+            ]];
+        }
+
+        if(empty($postChecksum) && empty($postContent)) {
+            $this->activityLog->setInfo('Empty content');
+            return ['error' => [
+                'message' => '投稿内容は空です'
+            ]];
+        }
+
+        $calculatedChecksum = Post::calculateChecksum($postContent);
+
+        if($postChecksum != $calculatedChecksum) {
+            $this->activityLog->setInfo('Checksum failed');
+            return ['error' => [
+                'message' => 'サーバへ送信する内容にエラーがあります'
+            ]];
+        }
+
+        $oldTitle = $post->title;
+
+        $post->title = !empty($postTitle) ? $postTitle : 'タイトルなし';
+        $post->content = !empty($postContent) ? $postContent : '&nbsp;';
+        $post->user_id = $this->currentUser->id;
+        $post->site_id = $request->get('site_id');
+        $post->category = $request->get('category');
+        $post->description = $postDescription;
+        $post->feature_img = $request->get('feature_img_url');
+        $post->priority = (int)$request->get('post_priority', 0);
+        $post->published_status = 0;
+        $post->word_count = (int)$request->get('word_count', 0);
+
+        $submitType = strpos($request->get('submit_type'), '公開') !== false ? 'publish' : 'draft';
+
+        if($submitType == 'publish' && !$post->canPublishBy($this->currentUser)) {
+            $submitType = 'draft';
+        }
+
+        if($submitType == 'publish' && $this->currentUser->is(User::CONTRIBUTOR)){
+            $this->activityLog->setInfo('Contributor can not publish');
+            return '404 Not Found';
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "404 Not Found" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)